rpcserver.go

```diff
@@ -3204,6 +3204,98 @@ func (r *rpcServer) DeleteFederationServer(ctx context.Context,
 	return &unirpc.DeleteFederationServerResponse{}, nil
 }
 
+// SetFederationSyncConfig sets the configuration of the universe federation
+// sync.
+func (r *rpcServer) SetFederationSyncConfig(ctx context.Context,
+	req *unirpc.SetFederationSyncConfigRequest) (
+	*unirpc.SetFederationSyncConfigResponse, error) {
+
+	// Update general sync config if set.
+	if req.GeneralSyncConfig != nil {
+		err := r.cfg.FederationDB.SetFederationGeneralSyncConfig(
+			ctx, req.GeneralSyncConfig.OnlySyncIssuanceProofs,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to set general sync "+
+				"config: %w", err)
+		}
+	}
+
+	// Update asset (asset/asset group) specific sync configs.
+	for i := range req.AssetSyncConfig {
+		assetSyncConfig := req.AssetSyncConfig[i]
+
+		// Parse the universe ID from the RPC form.
+		uniID, err := unmarshalUniID(assetSyncConfig.Id)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse universe id: %w",
+				err)
+		}
+
+		config := universe.FedUniSyncConfig{
+			UniverseID:             uniID,
+			Exclude:                assetSyncConfig.Exclude,
+			OnlySyncIssuanceProofs: assetSyncConfig.OnlySyncIssuanceProofs,
+		}
+		err = r.cfg.FederationDB.UpsertFederationUniSyncConfig(
+			ctx, config,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to set universe specific "+
```

Comment:
Print out which uni ID failed to apply?

Reply:
I wonder if we should try to make the config modification atomic as well. Reminds me of something we did on the lnd side: if ppl try to update routing configs, we tell them which of them applied vs not.